### PR TITLE
NO-JIRA: Use suggested functions instead of deprecated ones

### PR DESCRIPTION
**Pull request by ardaguclu**
With regards to healthiness of the code base of oc, this PR switches to some of suggested functions instead of the deprecated ones. 
This PR must be no-op.

### CODE REVIEW

File: pkg/cli/deployer/strategy/recreate/recreate.go [hunk 1]
```diff
@@ -253,7 +253,7 @@ func (s *RecreateDeploymentStrategy) scaleAndWait(deployment *corev1.Replication
 	// Wait for the scale to take effect.
 	if !alreadyScaled {
 		// FIXME: This should really be a watch, however the scaler client does not implement the watch interface atm.
-		err = wait.PollImmediate(1*time.Second, retryTimeout, func() (bool, error) {
+		err = wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, retryTimeout, true, func(ctx context.Context) (bool, error) {
```
Comment (flavianmissi) [inline pkg/cli/deployer/strategy/recreate/recreate.go:256]: this is unrelated to your changes, but is this `err` never handled?

**Comment by flavianmissi** [comment]
Nicely done 💪🏼 
Left a question related to previous code's error handling, will put a hold so it doesn't get merged until you make a decision on how to handle it - feel free to remove it.
/lgtm
/hold

File: pkg/cli/deployer/strategy/recreate/recreate.go (hunk 1 above)
Comment (ardaguclu) [reply pkg/cli/deployer/strategy/recreate/recreate.go:256]: Thank you for spending extra time reviewing this PR. You are right, it is not handled and needs to be fixed. But I'll have another refactoring PR soon (in probably next H&H) and I'll fix this in there.

**Comment by ardaguclu** [comment]
It is unclear whether it is related or not, better to try once more;
/test e2e-agnostic-ovn-cmd

**Pull request merged**